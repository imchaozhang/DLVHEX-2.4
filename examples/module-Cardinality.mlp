#module(p1, []). 
q(a). 
q(b).
r(a). 
r(b). 
s(a).
equalQR :- @p2[q,r]::equal.
notEqualQR :- not equalQR.
greater_thanRS :- @p2[r,s]::greater_than.
notGreaterThanRS :- not greater_thanRS.
less_thanSR :- @p2[s,r]::less_than.
notLessThanSR :- not less_thanSR.


#module(p2, [q1/1, q2/1]).
q1i(X) v q1i(Y) :- q1(X), q1(Y), X != Y.
q2i(X) v q2i(Y) :- q2(X), q2(Y), X != Y.
skip1 :- q1(X), not q1i(X).
skip2 :- q2(X), not q2i(X).
equal :- skip1 , skip2 , @p2[q1i,q2i]::equal.
equal :- not skip1 , not skip2.
less_than :- skip1 , skip2 , @p2[q1i,q2i]::less_than.
less_than :- not skip1 , skip2.
greater_than :- skip1 , skip2 , @p2[q1i,q2i]::greater_than.
greater_than :- skip1 , not skip2.
