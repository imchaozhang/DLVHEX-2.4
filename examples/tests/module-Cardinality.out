(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(a),q1i(a)}, p2[{q1(a),q2(a)}]={q1(a),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(a),q1i(a)}, p2[{q1(a),q2(a)}]={q1(a),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(a),q1i(a)}, p2[{q1(a),q2(a)}]={q1(a),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(a),q1i(a)}, p2[{q1(a),q2(a)}]={q1(a),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(a)}, p2[{q1(a),q2(b)}]={q1(a),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(a)}, p2[{q1(a),q2(b)}]={q1(a),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(a)}, p2[{q1(a),q2(b)}]={q1(a),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(a)}, p2[{q1(a),q2(b)}]={q1(a),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q1i(b),q2i(a)}, p2[{q1(b),q2(a)}]={q1(b),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q1i(b),q2i(a)}, p2[{q1(b),q2(a)}]={q1(b),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q1i(b),q2i(a)}, p2[{q1(b),q2(a)}]={q1(b),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q1i(b),q2i(a)}, p2[{q1(b),q2(a)}]={q1(b),q2(a),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(b)}, p2[{q1(b),q2(b)}]={q1(b),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(b)}, p2[{q1(b),q2(b)}]={q1(b),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(a)}, p2[{q1(a)}]={q1(a),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(b)}, p2[{q1(b),q2(b)}]={q1(b),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(a)}, p2[{q2(a)}]={q2(a),skip2,less_than})
(p1[{}]={q(a),q(b),r(a),r(b),s(a),equalQR,greater_thanRS,less_thanSR}, p2[{q1(a),q1(b),q2(a),q2(b)}]={q1(a),q1(b),q2(a),q2(b),skip1,skip2,equal,q2i(b),q1i(b)}, p2[{q1(b),q2(b)}]={q1(b),q2(b),skip1,skip2,equal}, p2[{}]={equal}, p2[{q1(a),q1(b),q2(a)}]={q1(a),q1(b),q2(a),skip1,skip2,greater_than,q1i(b)}, p2[{q1(b)}]={q1(b),skip1,greater_than}, p2[{q1(a),q2(a),q2(b)}]={q1(a),q2(a),q2(b),skip1,skip2,less_than,q2i(b)}, p2[{q2(b)}]={q2(b),skip2,less_than})
